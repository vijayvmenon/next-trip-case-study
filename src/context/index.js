import { useContext } from 'react';
import AppContext, { Context } from './context';

export default AppContext;
export { Context };
export const useMyContext = () => useContext(Context);
