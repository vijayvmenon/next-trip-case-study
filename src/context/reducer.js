import constants from '../constants';
import ENTranslations from '../locales/en.json';
import { localStorageExists } from '../utils';

const localStorageLanguage =
  localStorageExists && localStorage.getItem('LANGUAGE');
const initialState = {
  tableExpanded: false,
  hideHeader: false,
  snackbar: { value: '', text: '' },
  language: localStorageLanguage || 'en',
  translations: localStorageLanguage
    ? constants.LANGUAGE_MAPPING[localStorageLanguage]
    : ENTranslations,
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case 'TABLE_EXPAND':
      return { ...state, tableExpanded: true };
    case 'TABLE_COLLAPSE':
      return { ...state, tableExpanded: false };
    case 'SET_SNACKBAR':
      return { ...state, snackbar: { value: action.value, text: action.text } };
    case 'HIDE_HEADER':
      return { ...state, hideHeader: true };
    case 'SHOW_HEADER':
      return { ...state, hideHeader: false };
    case 'SET_LANGUAGE':
      return {
        ...state,
        language: action.value,
        translations: constants.LANGUAGE_MAPPING[action.value],
      };
    default:
      return { ...state };
  }
}

export { initialState };
