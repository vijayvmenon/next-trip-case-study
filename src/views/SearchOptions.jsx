import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
import LinkButton from '../components/LinkButton';
// import constants from '../constants';

const useStyles = createUseStyles({
  root: {
    display: 'flex',
    justifyContent: 'center',
    position: 'relative',
  },
});

export default function SearchOptions({ options, click }) {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      {options.map((val) => (
        <LinkButton
          name={val.name}
          hrefLink={val.hrefLink}
          active={val.active}
          key={val.name}
          click={click}
        />
      ))}
    </div>
  );
}

SearchOptions.propTypes = {
  options: PropTypes.arrayOf(PropTypes.any).isRequired,
  click: PropTypes.func.isRequired,
};
