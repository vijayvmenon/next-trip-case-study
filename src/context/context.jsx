import React, { useReducer, createContext } from 'react';
import PropTypes from 'prop-types';
import reducer, { initialState } from './reducer';

export const Context = createContext();

export default function AppContext({ children }) {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
}

AppContext.propTypes = {
  children: PropTypes.objectOf(PropTypes.any).isRequired,
};
