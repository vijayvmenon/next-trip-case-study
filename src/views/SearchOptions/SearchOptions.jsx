import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
import LinkButton from '../../components/UI/LinkButton';
import { Context } from '../../context';

const useStyles = createUseStyles({
  root: {
    display: 'flex',
    justifyContent: 'center',
    position: 'relative',
  },
});

export default function SearchOptions({ options }) {
  const classes = useStyles();
  const { state: { translations = {} } = {} } = useContext(Context);

  return (
    <nav className={classes.root}>
      {options.map((val) => (
        <LinkButton
          name={translations[val.name]}
          hrefLink={val.hrefLink}
          key={val.name}
        />
      ))}
    </nav>
  );
}

SearchOptions.propTypes = {
  options: PropTypes.arrayOf(PropTypes.any).isRequired,
};
