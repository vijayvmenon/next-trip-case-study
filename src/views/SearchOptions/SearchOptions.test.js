import { render, fireEvent, screen } from '@testing-library/react';
import SearchOptions from './SearchOptions';
import renderer from 'react-test-renderer';
import { Context } from '../../context';

const dispatch = jest.fn(); // mock dispatch

const props = {
  options: [
    {
      name: 'BY ROUTE',
      hrefLink: '/byRoute',
    },
    {
      name: 'BY STOP',
      hrefLink: '/byStop',
    },
  ],
};

describe('Component:SearchOptions.jsx', () => {
  test('SearchOptions.jsx renders correctly', () => {
    const searchOptions = renderer
      .create(
        <Context.Provider value={{ dispatch }}>
          <SearchOptions {...props} />
        </Context.Provider>,
      )
      .toJSON();
    expect(searchOptions).toMatchSnapshot();
  });
});
