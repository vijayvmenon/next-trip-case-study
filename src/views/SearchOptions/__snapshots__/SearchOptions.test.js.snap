// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:SearchOptions.jsx SearchOptions.jsx renders correctly 1`] = `
<nav
  className="root-0-2-1"
>
  <a
    className="linkButtonRoot-0-2-2 undefined"
    data-testid="custom-anchor-"
    href="/byRoute"
    onClick={[Function]}
    to={null}
  >
    
  </a>
  <a
    className="linkButtonRoot-0-2-2 undefined"
    data-testid="custom-anchor-"
    href="/byStop"
    onClick={[Function]}
    to={null}
  >
    
  </a>
</nav>
`;
